15-Dec-2022 18:15:06.421 INFO [localhost-startStop-1] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
15-Dec-2022 18:15:06.421 INFO [localhost-startStop-1] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
15-Dec-2022 18:15:06.423 INFO [localhost-startStop-1] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@2b12f019')
15-Dec-2022 19:57:39.031 INFO [localhost-startStop-2] org.apache.catalina.core.ApplicationContext.log SessionListener: contextDestroyed()
15-Dec-2022 19:57:39.031 INFO [localhost-startStop-2] org.apache.catalina.core.ApplicationContext.log ContextListener: contextDestroyed()
15-Dec-2022 21:35:48.730 INFO [localhost-startStop-1] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
15-Dec-2022 21:35:48.730 INFO [localhost-startStop-1] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
15-Dec-2022 21:35:48.732 INFO [localhost-startStop-1] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@127abd77')
15-Dec-2022 22:37:28.827 SEVERE [http-nio-8080-exec-58] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/num] threw exception
	java.io.IOException: An exception occurred processing [/index.jsp] at line [27]

24:       <% 
25:         DataSQL data = new DataSQL();
26:         
27:         data.TestCallJAVASCRIPT();
28:         
29:         List<User> users = data.selectAllUsers();
30:         


Stacktrace:
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:493)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:383)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:331)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:698)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:367)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:639)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:882)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1647)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.base/java.lang.Thread.run(Thread.java:833)
	Caused by: java.nio.file.NoSuchFileException: ..\..\web\main.js
		at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
		at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
		at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
		at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:236)
		at java.base/java.nio.file.Files.newByteChannel(Files.java:380)
		at java.base/java.nio.file.Files.newByteChannel(Files.java:432)
		at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:422)
		at java.base/java.nio.file.Files.newInputStream(Files.java:160)
		at java.base/java.nio.file.Files.newBufferedReader(Files.java:2923)
		at num.DataSQL.TestCallJAVASCRIPT(DataSQL.java:65)
		at org.apache.jsp.index_jsp._jspService(index_jsp.java:150)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:465)
		... 25 more
15-Dec-2022 22:38:19.340 SEVERE [http-nio-8080-exec-60] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/num] threw exception
	java.io.IOException: An exception occurred processing [/index.jsp] at line [27]

24:       <% 
25:         DataSQL data = new DataSQL();
26:         
27:         data.TestCallJAVASCRIPT();
28:         
29:         List<User> users = data.selectAllUsers();
30:         


Stacktrace:
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:493)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:383)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:331)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:698)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:367)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:639)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:882)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1647)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.base/java.lang.Thread.run(Thread.java:833)
	Caused by: java.nio.file.NoSuchFileException: ..\..\..\web\main.js
		at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
		at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
		at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
		at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:236)
		at java.base/java.nio.file.Files.newByteChannel(Files.java:380)
		at java.base/java.nio.file.Files.newByteChannel(Files.java:432)
		at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:422)
		at java.base/java.nio.file.Files.newInputStream(Files.java:160)
		at java.base/java.nio.file.Files.newBufferedReader(Files.java:2923)
		at num.DataSQL.TestCallJAVASCRIPT(DataSQL.java:65)
		at org.apache.jsp.index_jsp._jspService(index_jsp.java:150)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:465)
		... 25 more
15-Dec-2022 22:39:17.473 SEVERE [http-nio-8080-exec-58] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/num] threw exception [An exception occurred processing [/index.jsp] at line [27]

24:       <% 
25:         DataSQL data = new DataSQL();
26:         
27:         data.TestCallJAVASCRIPT();
28:         
29:         List<User> users = data.selectAllUsers();
30:         


Stacktrace:] with root cause
	java.lang.NullPointerException: Cannot invoke "javax.script.ScriptEngine.eval(java.io.Reader)" because "engine" is null
		at num.DataSQL.TestCallJAVASCRIPT(DataSQL.java:65)
		at org.apache.jsp.index_jsp._jspService(index_jsp.java:150)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:465)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:383)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:331)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:698)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:367)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:639)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:882)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1647)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.base/java.lang.Thread.run(Thread.java:833)
15-Dec-2022 22:41:01.956 SEVERE [http-nio-8080-exec-81] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/num] threw exception [Unable to compile class for JSP: 

An error occurred at line: [36] in the jsp file: [/index.jsp]
System.out cannot be resolved to a type
33:         ScriptEngineManager sem = new ScriptEngineManager();
34: List<ScriptEngineFactory> factories = sem.getEngineFactories();
35: for (ScriptEngineFactory factory : factories)
36:     System.out.println(factory.getEngineName() + " " + factory.getEngineVersion() + " " + factory.getNames());
37: if (factories.isEmpty())
38:     System.out.println("No Script Engines found");
39:         


An error occurred at line: [38] in the jsp file: [/index.jsp]
System.out cannot be resolved to a type
35: for (ScriptEngineFactory factory : factories)
36:     System.out.println(factory.getEngineName() + " " + factory.getEngineVersion() + " " + factory.getNames());
37: if (factories.isEmpty())
38:     System.out.println("No Script Engines found");
39:         
40:         List<User> users = data.selectAllUsers();
41:         


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [36] in the jsp file: [/index.jsp]
System.out cannot be resolved to a type
33:         ScriptEngineManager sem = new ScriptEngineManager();
34: List<ScriptEngineFactory> factories = sem.getEngineFactories();
35: for (ScriptEngineFactory factory : factories)
36:     System.out.println(factory.getEngineName() + " " + factory.getEngineVersion() + " " + factory.getNames());
37: if (factories.isEmpty())
38:     System.out.println("No Script Engines found");
39:         


An error occurred at line: [38] in the jsp file: [/index.jsp]
System.out cannot be resolved to a type
35: for (ScriptEngineFactory factory : factories)
36:     System.out.println(factory.getEngineName() + " " + factory.getEngineVersion() + " " + factory.getNames());
37: if (factories.isEmpty())
38:     System.out.println("No Script Engines found");
39:         
40:         List<User> users = data.selectAllUsers();
41:         


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:102)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:213)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:569)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:380)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:350)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:334)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:597)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:398)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:383)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:331)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:698)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:367)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:639)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:882)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1647)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.base/java.lang.Thread.run(Thread.java:833)
